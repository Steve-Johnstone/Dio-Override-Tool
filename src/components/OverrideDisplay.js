import React from 'react';
import PropTypes from 'prop-types';
import 'bulma/css/bulma.min.css';

const OverrideDisplay = ({ selectedOverrides }) => {
	return (
		<section className='section mt-6'>
			<div className='container'>
				<h3 className='title is-3'>All Overrides</h3>
				<div>
					<ul id='override-display'>
						{selectedOverrides.map((a) => {
							return <li key={selectedOverrides.indexOf(a)}>{a}</li>;
						})}
					</ul>
				</div>
			</div>
		</section>
	);
};

OverrideDisplay.propTypes = {
	selectedOverrides: PropTypes.array,
};

export default OverrideDisplay;
