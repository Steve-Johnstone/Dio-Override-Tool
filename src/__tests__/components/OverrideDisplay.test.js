import { render, screen, within } from '@testing-library/react';
import renderer from 'react-test-renderer';
import OverrideDisplay from '../../components/OverrideDisplay';
import { MemoryRouter } from 'react-router-dom';

describe('OverrideDisplay component', () => {
	const component = (
		<MemoryRouter>
			<OverrideDisplay
				url=''
				overrideList={[
					'Australian Print Receipt',
					'Consolidated Amenities',
					'Event Widget',
					'Insurance Details',
					'Multi Room',
					'No Cancel Button',
					'No Guest Reviews',
					'United Only Points',
				]}
			/>
		</MemoryRouter>
	);

	it('should display the DOM correctly', () => {
		const tree = renderer.create(component).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render the component correctly, based on the props passed to it', () => {
		render(component);

		//Check that the overridelist is rendered
		const list = screen.getByTestId('override-list');
		expect(list).toBeInTheDocument();

		//Check that the list contains 8 items
		const { getAllByRole } = within(list);
		const items = getAllByRole('listitem');
		expect(items.length).toBe(8);

		//Check that list includes the selected overrides
		expect(screen.getByText('Australian Print Receipt')).toBeInTheDocument();
		expect(screen.getByText('No Cancel Button')).toBeInTheDocument();
	});
});
