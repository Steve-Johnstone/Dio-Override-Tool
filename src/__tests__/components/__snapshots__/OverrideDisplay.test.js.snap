// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverrideDisplay component should display the DOM correctly 1`] = `
<div
  className="override-display-conatiner"
>
  <div
    className="header"
  >
    <div
      className="level-item is-justify-content-space-between"
    >
      <div
        className="is-flex is-justify-content-space-between"
      >
        <img
          alt="hcom logo"
          className="logo"
          data-testid="logo"
          src="image/hotelscom-logo.png"
        />
        <p>
          Dio Override Tool
        </p>
      </div>
      <div>
        <form
          className="field py-5"
          id="search-bar-container"
        >
          <div
            className="search-bar"
          >
            <div
              className="search-icon"
              data-testid="search-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                />
              </svg>
            </div>
            <input
              className="input"
              id="search-input-field"
              name="search-item"
              onChange={[Function]}
              placeholder="Start typing to filter..."
              type="text"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="container p-5"
  >
    <h4
      className="title is-4"
    >
      All Overrides
    </h4>
    <div>
      <ul
        className="override-list"
        data-testid="override-list"
      >
        <li
          className=""
          onClick={[Function]}
        >
          Australian Print Receipt
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Consolidated Amenities
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Event Widget
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Insurance Details
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          Multi Room
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          No Cancel Button
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          No Guest Reviews
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          United Only Points
        </li>
      </ul>
    </div>
  </div>
  <div
    className="footer-section"
  >
    <div
      className="footer-override-display"
    >
      <ul
        className="footer-override-list"
      />
    </div>
    <div
      className="footer-url-display ml-5 level-item is-justify-content-flex-start"
    >
      <div
        className="ml-3"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
          />
        </svg>
      </div>
      <div
        className="p-2"
        data-testid="url-display"
        id="url-display"
      >
        localhost:8080/?override=
      </div>
    </div>
    <div
      className="level-item is-justify-content-space-between ml-5"
    >
      <div>
        Click anywhere on the box to select the whole URL, this way you can copy it.
      </div>
      <div
        className="mr-5"
      >
        <button
          className="button is-white m-2"
          name="clear-all"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
            />
          </svg>
            Clear All
        </button>
        <a
          className="button m-2"
          href="localhost:8080/?override="
          id="go-button"
          name="show-it"
          target="blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 388c-72.597 0-132-59.405-132-132 0-72.601 59.403-132 132-132 36.3 0 69.299 15.4 92.406 39.601L278 234h154V80l-51.698 51.702C348.406 99.798 304.406 80 256 80c-96.797 0-176 79.203-176 176s78.094 176 176 176c81.045 0 148.287-54.134 169.401-128H378.85c-18.745 49.561-67.138 84-122.85 84z"
            />
          </svg>
            Show it
        </a>
      </div>
    </div>
  </div>
</div>
`;
